WordPress/pattern-directory
# Pattern Administration: Add a modal to capture the reason for unlisting a pattern

Fixes #105 — Add a modal like the one used for reporting a pattern, which can capture a generic reason for unlisting, and a space for an internal note which will show up in the Internal Notes sidebar.

The reason ID is saved into a new post meta field, `wpop_unlisted_reason`, which can be used in the theme for #371 Showing a reason for unlisting to the pattern author.

### Screenshots

The "Unlist" button is the same, but it brings up a modal now.
<img width="298" alt="Screen Shot 2021-11-11 at 12 57 34 PM" src="https://user-images.githubusercontent.com/541093/141346793-5c8d1644-c88d-4626-abb9-dd608cd8b5b9.png">
<img width="539" alt="Screen Shot 2021-11-11 at 12 58 16 PM" src="https://user-images.githubusercontent.com/541093/141346796-03174f71-22ec-44fd-b214-0d62ec271454.png">

Once submitted, the note is saved to internal notes, and the reason's term ID is saved to the pattern's post meta in `wpop_unlisted_reason`.
<img width="558" alt="Screen Shot 2021-11-11 at 1 07 53 PM" src="https://user-images.githubusercontent.com/541093/141347675-208c50e1-7776-4656-8d9d-e3994cb6912c.png">

The reason and note are visible in the Internal Notes sidebar.
<img width="304" alt="Screen Shot 2021-11-11 at 1 06 35 PM" src="https://user-images.githubusercontent.com/541093/141347477-ec7ead07-3b75-4b44-9e0f-8c9306c14e2e.png">

### How to test the changes in this Pull Request:

1. Open a pattern in the wp-admin editor
2. Click the "Unlist" button in the sidebar
3. A modal should pop up, the reasons will load in after a second
4. Fill out the modal, submit it
5. It should save without error, and the pattern should now be marked as unlisted
6. Reload the editor, and open the Internal Notes sidebar
7. Your note should be visible there, with your username and the date.
